editor/plugins/item_list_editor_plugin.cpp

```diff
@@ -143,22 +150,46 @@ int ItemListOptionButtonPlugin::get_flags() const {
 	return FLAG_ICON | FLAG_ID | FLAG_ENABLE;
 }
 
-void ItemListOptionButtonPlugin::add_item() {
-	ob->add_item(vformat(TTR("Item %d"), ob->get_item_count()));
+void ItemListOptionButtonPlugin::undoRedo_add_new_item(int item_idx) {
+	ob->add_item(vformat(TTR("Item %d"), item_idx));
+	_change_notify();
+}
+
+void ItemListOptionButtonPlugin::undoRedo_re_add_item(unsigned int position, String name, Ref<Texture2D> icon, bool enabled){
+	ob->add_item(name);
 	_change_notify();
 }
 
+void ItemListOptionButtonPlugin::undoRedo_erase(int item_idx){
+	ob->remove_item(item_idx);
+	_change_notify();
+}
+
+void ItemListOptionButtonPlugin::add_item() {
+	undo_redo->create_action(TTR("ItemListOptionButtonPlugin Add Element"));
+	int item_idx = ob->get_item_count();
+    undo_redo->add_do_method(this, "undoRedo_add_new_item", item_idx);
+    undo_redo->add_undo_method(this, "undoRedo_erase", item_idx);
+    undo_redo->commit_action();
+}
+
 int ItemListOptionButtonPlugin::get_item_count() const {
 	return ob->get_item_count();
 }
 
 void ItemListOptionButtonPlugin::erase(int p_idx) {
-	ob->remove_item(p_idx);
-	_change_notify();
+	String name = get_item_text(p_idx);
+	Ref<Texture2D> icon = get_item_icon(p_idx);
+	bool is_enabled = is_item_enabled(p_idx);
+	undo_redo->create_action(TTR("ItemListItemListPlugin Remove Element"));
```

Comment:
I think the action text could be better. Maybe something like "Removed Item 3 from Item List", where "3" is the p_idx. ItemListItemListPlugin does not mean anything to the Godot user - it is an internal editor class.